rotkehlchen/globaldb/handler.py

```diff
@@ -1347,6 +1349,36 @@ def soft_reset_assets_list() -> Tuple[bool, str]:
         cursor.execute(detach_database)
         return True, ''
 
+    def save_binance_pairs(
+        self,
+        new_pairs: Iterable['BinancePair'],
+        location: Location,
+    ) -> bool:
+        query = 'INSERT OR IGNORE INTO binance_pairs(pair, base_asset, quote_asset, location) VALUES (?, ?, ?, ?)'  # noqa: E501
+        cursor = self._conn.cursor()
+        try:
+            cursor.executemany(query, [pair.serialize_for_db() for pair in new_pairs])
+            self.add_setting_value(name=f'binance_pairs_queried_at_{location}', value=ts_now())
+            return True
+        except sqlite3.IntegrityError as e:
+            log.error(f'Failed to store binance pairs in database. {str(e)}')
+        return False
+
+    def get_binance_pairs(self, location: Location) -> List['BinancePair']:
+        # TODO: Change the logic around BinancePair to avoid the importing here
+        from rotkehlchen.exchanges.utils import BinancePair  # noqa: E501  # pylint: disable=import-outside-toplevel  # isort:skip
+        cursor = self._conn.cursor()
+        cursor.execute(
+            'SELECT pair, base_asset, quote_asset, location FROM binance_pairs WHERE location=?',
+            location.serialize_for_db(),
+        )
+        try:
+            pairs = [BinancePair.deserialize_from_db(pair) for pair in cursor]
+        except DeserializationError as e:
+            log.debug(f'Failed to deserialize binance pairs. {str(e)}')
+            return []
```

Comment:
You are overusing list comprehensions to the point it hurts you.

Switch this to a normal for loop. And instead of a single error stopping the entire deserialization make it so that an error is just skipping that particular pair but all other pairs are fine.